homeassistant/components/aqara/binary_sensor.py

```diff
@@ -0,0 +1,221 @@
+"""Support for Aqara binary sensors."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Any
+
+from aqara_iot import AqaraDeviceManager, AqaraPoint
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity import EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import HomeAssistantAqaraData
+from .base import AqaraEntity, find_aqara_device_points_and_register
+from .const import AQARA_BATTERY_LOW_ENTITY_NEW, AQARA_DISCOVERY_NEW, DOMAIN
+
+
+@dataclass
+class AqaraBinarySensorEntityDescription(BinarySensorEntityDescription):
+    """Describes a Aqara binary sensor."""
+
+    # Value to consider binary sensor to be "on"
+    on_value: Any = True
+    on_icon: str = "mdi:restart"
+    off_icon: str = "mdi:restart"
+
+    def set_key(self, key) -> AqaraBinarySensorEntityDescription:
+        """Set key of binary Description."""
+        self.key = key
+        return self
+
+    def set_name(self, name) -> AqaraBinarySensorEntityDescription:
+        """Set name of binary Description."""
+        self.name = name
+        return self
+
+    def set_icon(
+        self, default_icon: str, on_icon: str, off_icon: str
+    ) -> AqaraBinarySensorEntityDescription:
+        """Set name of binary Description."""
+        self.icon = default_icon
+        self.on_icon = on_icon
+        self.off_icon = off_icon
+        return self
+
+
+battery_staus_desc = AqaraBinarySensorEntityDescription(  # 电池电量情况，1低电量，0恢复电量
```

Comment:
Please keep all code and comments to English.

Reply:
I suppose maybe an exception could be made for device names. But that's a big maybe IMO.

Reply:
No, there is no exception. Everybody in this project has to be able to read and understand all code & comments. The use of the English language is required.